### PR TITLE
feat:change

### PR DESCRIPTION
**Pull request by big-dust**
change

### CODE REVIEW

File: handler/post/post.go [hunk 1]
```diff
@@ -180,91 +169,60 @@ func CreatePost(c *gin.Context) {
...
+	//	UserId:  service.GetId(c),
+	//	Content: m["content"],
+	//	Status:  1,
+	//	Private: true/false,
+	//}
+	var comment model.Comment
+	err1 := c.ShouldBind(&comment)
+	comment.UserId = service.GetId(c)
+	err2 := model.CreateComment(comment)
+	if err1 != nil || err2 != nil {
```
Comment (Bowser1704) [inline handler/post/post.go:244]: Maybe add some `logger` to print this err will be better.

File: service/whether_my.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ func WhetherMyPost(userId interface{}, postId interface{}) bool {
 	return false
 }
 
-func WhetherMyComment(userId interface{}, commentId interface{}) bool {
+func IsMyComment(userId interface{}, commentId interface{}) bool {
```
Comment (Bowser1704) [inline service/whether_my.go]: Interface here is no need that I can see. Can you add some comment?

File: router/router.go [hunk 3]
```diff
@@ -16,52 +16,52 @@ func Register(r *gin.Engine) {
...
-		p.PUT("", post.UpdatePost)                   //修改帖子 
-		p.POST("/comment", post.AddComment)          //在帖子下面发评论  
+		p.POST("", post.CreatePost)                  //发布帖子
+		p.DELETE("", post.DeletePost)                //删除帖子
+		p.PUT("", post.UpdatePost)                   //修改帖子
+		p.POST("/comment", post.AddComment)          //在帖子下面发评论
 		p.DELETE("/comment", post.DeleteComment)     //删除评论
-		p.POST("/comment_reply", post.AddReply)      //在评论下回复  
+		p.POST("/comment_reply", post.AddReply)      //在评论下回复
 		p.DELETE("/comment_reply", post.DeleteReply) //删除评论的回复
```
Comment (Bowser1704) [inline router/router.go:57]: IMO, The `/my_posts` and `/:id/user_post` can be merged to `/:id/posts`. So are other APIs.

File: handler/post/post.go (hunk 1 above)
Comment (big-dust) [reply handler/post/post.go:244]: > Maybe add some `logger` to print this err will be better.
Yes.

File: service/whether_my.go (hunk 2 above)
Comment (big-dust) [reply service/whether_my.go]: > Interface here is no need that I can see. Can you add some comment?

确实不需要，偷懒直接用这个了。

File: router/router.go (hunk 3 above)
Comment (big-dust) [reply router/router.go:57]: > IMO, The `/my_posts` and `/:id/user_post` can be merged to `/:id/posts`. So are other APIs.

嗯

File: handler/post/post.go [hunk 4]
```diff
@@ -6,50 +6,47 @@ import (
...
 // @Param  start  query integer true "开始点"
 // @Param  length query integer true "需要返回的帖子数量"
-// @Success 200 {object} handler.Response "{"msg":"查询到最新的帖子（数组）"}"
+// @Success 200 {object} []handler.Post "{"msg":"查询到最新的帖子（数组）"}"
 // @Failure 410 {object} handler.Error  "{"msg":"最新帖子查询失败"}"
 // @Router /post/latest [GET]
 func Latest(c *gin.Context) {
-	posts, err := model.QueryNewPosts(c.Query("start"), c.Query("length"))
+	start, _ := strconv.Atoi(c.Query("start"))
+	length, _ := strconv.Atoi(c.Query("length"))
```
Comment (Bowser1704) [inline handler/post/post.go]: This error cannot be ignored.

`start` will be empty value if someone pass an invalid int parameter.

File: handler/struct.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	FilePath1 string    `form:"file_path1" json:"file_path1"`
+	FilePath2 string    `form:"file_path2" json:"file_path2"`
+	FilePath3 string    `form:"file_path3" json:"file_path3"`
+	FilePath4 string    `form:"file_path4" json:"file_path4"`
+	FilePath5 string    `form:"file_path5" json:"file_path5"`
+	FilePath6 string    `form:"file_path6" json:"file_path6"`
+}
+
+type Relationship struct {
+	ID         uint `gorm:"primarykey"`
```
Comment (Bowser1704) [inline handler/struct.go:72]: Use `gorm.Entity` can simplify this code.

**Pull request merged**